docker_cli/_image.py

```diff
@@ -0,0 +1,451 @@
+import io
+import os
+from shlex import split
+from subprocess import DEVNULL, PIPE, CalledProcessError, run
+from sys import stdin
+from typing import (Any, Iterable, List, Optional, Type, TypeVar, Union,
+                    overload)
+
+from ._exceptions import (CommandNotFoundError, DockerBuildError,
+                          ImageNotFoundError)
+
+
+class Image:
+    """
+    A docker image.
+
+    Holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
+
+    Capabilities include:
+    -   Building docker images from dockerfiles or dockerfile-formatted strings
+        via :func:`~docker_cli.Image.build`.
+    -   Running commands in containers built from the image using
+        :func:`~docker_cli.Image.run`.
+    -   Inspecting properties of the given image.
+    """
+    Self = TypeVar("Self", bound="Image")
+
+    def __init__(self, name_or_id: str):
+        """
+        Initialize a new Image object.
+
+        Parameters
+        ----------
+        name_or_id : str
+            A name or ID by which to find this image using docker inspect.
+        """
+        self._id = get_image_id(name_or_id)
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        tag: str,
+        *,
+        dockerfile: Optional[os.PathLike[str]],
+        context: Union[str, os.PathLike[str]],
+        stdout: Optional[io.TextIOBase],
+        stderr: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ) -> Self:
+        """
+        Build a new image from a dockerfile.
+
+        Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        tag : str
+            A name for the image.
+        dockerfile : os.PathLike
+            The path of the Dockerfile to build, relative to the `context`
+            directory.
```

Comment:
We tested this out (specifying the Dockerfile path relative to the context directory) and weren't able to get it to work for some reason, so we should update the docstring so we don't mislead users.

https://github.com/isce3-testing/image-sandbox/pull/1#discussion_r1140442200

```suggestion
            The path of the Dockerfile to build.
```